File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    var answer = [];
    
    let sums = [0];
    
    sequence.forEach((item, i) => sums.push(item + sums[i]));
    let len = sums.length;
    
    let maxLen = 1000001;
    let startIdx =0;
    let endIdx =1;
    
    while(startIdx <= endIdx && endIdx < len) {
        let item = sums[endIdx] - sums[startIdx];
        
        if(item === k) {
            let curLen = endIdx - startIdx-1;
            if(maxLen > curLen) {
                answer = [startIdx, endIdx-1];
                maxLen = curLen;
            }
        } 
        
        if (item > k) startIdx +=1;
        else endIdx +=1;
    }
    
    return answer;
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> sequence, int k) {
    vector<int> answer;
    
    vector<int> sums = {0};
    
    int len = sequence.size();
    
    for(int i=0; i<len ;i++) 
    {
        sums.push_back(sums[i] + sequence[i]);
    }
    
    int maxLen = 1000001;
    int startIdx = 0;
    int endIdx = 1;
    
    while(endIdx <= len) 
    {
        int value = sums[endIdx] - sums[startIdx];
        
        if(value == k) 
        {
            int curLen = endIdx - startIdx -1;
            
            if(maxLen > curLen)
            {
                answer = {startIdx, endIdx -1};
                maxLen =curLen;
            }
        }
        
        if(value < k) endIdx +=1;
        else startIdx +=1;
    }
    
    return answer;
}